### PR TITLE
Skips the test if `import bitsandbytes` throws error

### PR DESCRIPTION
**Pull request by kiya00**
<details>
  <summary><b>Before submitting</b></summary>

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/main/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

</details>

## What does this PR do?

According to https://github.com/bitsandbytes-foundation/bitsandbytes/blob/main/docs/source/installation.mdx, bitsandbytes is currently only supported on CUDA GPUs for CUDA versions 11.0 - 12.6. When CUDA>=12.7 is installed the `package_available` returns True but `import bitsandbytes ` fails, so we skip the test in both `package_available=False || import bitsandbytes fails`
 
cc: @t-vi

### CODE REVIEW

**Review by crcrpar — APPROVED** [review]

File: thunder/tests/framework.py [hunk 1]
```diff
@@ -62,6 +62,19 @@ class NOTHING:
 IS_WINDOWS = platform.system() == "Windows"
 
 
+def _bitsandbytes_available():
+    if not package_available("bitsandbytes"):
+        return False
+    try:
+        import bitsandbytes
+    except Exception:
```
Comment (t-vi) [inline thunder/tests/framework.py:70]: Thank you @kiya00. Would we want to make this more specific, eg `(ImportError, RuntimeError)` like below?

Comment (kiya00) [reply thunder/tests/framework.py]: Thank you @t-vi , I've added it

**Comment by kiya00** [comment]
Hi @t-vi , I think it's ready to merge

**Review by t-vi — APPROVED** [review]
Thank you @kiya00

**Pull request merged**